test/test_jit.py

```diff
@@ -10517,19 +10511,25 @@ def test_rand():
         out = fn()
         self.assertEqual(out.dtype, torch.double)
         # Testing shape analysis correctly setting type
+
+        # this only works because TE sets types on values
+        # otherwise we would erase all prim::profile nodes
         FileCheck().check("Double(3:4, 4:1, requires_grad=0, device=cpu)") \
                    .check_not("Float(3:4, 4:1, requires_grad=0, device=cpu)").run(fn.graph_for())
 
         @torch.jit.script
         def randint():
             return torch.randint(0, 5, [1, 2])
 
-        out = randint(profile_and_replay=True)
-        self.assertEqual(out.dtype, torch.double)
         # although the type should be int here, testing that the runtime dtype
         # and shape analysis dtype is the same.
-        FileCheck().check("Double(1:2, 2:1, requires_grad=0, device=cpu)") \
-                   .check_not("Float(1:2, 2:1, requires_grad=0, device=cpu)").run(randint.graph_for())
+        with enable_profiling_mode_for_profiling_tests():
```

Comment:
In the future, tests that are only run with the profiling executor should be in a separate file